src/python/dxpy/bindings/dxfile.py

```diff
@@ -363,6 +396,16 @@ def write(self, data, multithread=True, **kwargs):
             does not affect where the next :meth:`write` will occur.
 
         '''
+        self.file_limits = dxpy.api.project_describe(self.project, {'fields': {'fileUploadParameters': True}})['fileUploadParameters']
+        self._set_file_limits(self.file_limits)
+
+        self._write_buf_size = self._get_buffer_size_for_file(self.file_size, self.file_is_mmapd)
```

Comment:
get_buffer_size_for_file could just ensure that we respect the minimum and maximum part sizes.

I also think that instead of throwing an error, we should just adjust the buffer size to comply with the maximum/minimum (selecting the nearest legitimate value, and possibly not using the exact part size the client wanted). Otherwise, a part size that the client sets for Azure will fail on AWS, and vice versa.